cpu/atmega_common/periph/gpio.c

```diff
@@ -241,38 +241,38 @@ int gpio_init(gpio_t pin, gpio_mode_t mode)
     return 0;
 }
 
-int gpio_read(gpio_t pin)
+int gpio_cpu_read(gpio_t pin)
 {
     return (_SFR_MEM8(_pin_addr(pin)) & (1 << _pin_num(pin)));
 }
 
-void gpio_set(gpio_t pin)
+void gpio_cpu_set(gpio_t pin)
 {
     _SFR_MEM8(_port_addr(pin)) |= (1 << _pin_num(pin));
 }
 
-void gpio_clear(gpio_t pin)
+void gpio_cpu_clear(gpio_t pin)
 {
     _SFR_MEM8(_port_addr(pin)) &= ~(1 << _pin_num(pin));
 }
 
-void gpio_toggle(gpio_t pin)
+void gpio_cpu_toggle(gpio_t pin)
 {
-    if (gpio_read(pin)) {
-        gpio_clear(pin);
+    if (gpio_cpu_read(pin)) {
+        gpio_cpu_clear(pin);
     }
     else {
-        gpio_set(pin);
+        gpio_cpu_set(pin);
     }
 }
```

Comment:
Unrelated, but wouldn't this be both easier to read and more efficient?

```C
_SFR_MEM8(_port_addr(pin)) ^= ~(1 << _pin_num(pin));
```

Reply:
Probably yes, but this was the former implementation and I didn't want to change anything else than the names of the low level function to avoid side effects.